frontend/src/app/modules/matches/matches-saga.test.js

```diff
@@ -0,0 +1,55 @@
+import { addMatch } from './matches-effects'
+import { addMatchSaga } from './matches-saga'
+import { call, put, select } from 'redux-saga/effects'
+import { MatchesActions } from './matches-actions'
+import { inProgress, success, failure } from '../api/request-status'
+import { getUsersSaga } from '../users/users-saga'
+import { getSelectedGame } from '../games/games-selectors'
+
+const MATCH = {}
+const FOOSBALL = { name: 'foosball' }
+const ERROR = {
+  message: 'ERROR',
+}
+
+describe('matchesSaga', () => {
+  describe('addMatchSaga', () => {
+    let gen
+    const expectNextToBe = (value, done = false) => {
+      const result = gen.next()
+      expect(result.value).toStrictEqual(value)
+      expect(result.done).toBe(done)
+    }
+    const expectDone = () => {
+      expectNextToBe(undefined, true)
+    }
+    beforeEach(() => {
+      gen = addMatchSaga({
+        match: MATCH,
+      })
+    })
+    describe('when addMatch is successful', () => {
+      it('performs correctly the sequence', () => {
+        expectNextToBe(select(getSelectedGame))
+        expect(gen.next(FOOSBALL).value)
```

Comment:
You could theoretically use expectNextToBe(expectedResult, suppliedValue=undefined) here.
where expectedResult could be an object having optional `value` and `done` properties which would be check in case of existence

Reply:
Fixed in https://github.com/salsita/foosball-rating/pull/125/commits/87707178781b189c0157f2314e3bb531c773c7c0

Reply:
I merged yours and mine suggestions :)